File: frontend/src/types/vendor.ts
export interface Vendor {
  id: number;
  name: string;
  address?: string | null;
  is_active: boolean;
  created_at?: string | null;
}

export interface VendorCreateRequest {
  name: string;
  address?: string;
}

export interface VendorUpdateRequest {
  name: string;
  address?: string;
}

File: api/schemas/vendor.py
"""Vendor API schemas."""
from typing import Optional

from pydantic import BaseModel, Field


class VendorResponse(BaseModel):
    """Active vendor record."""

    id: int
    name: str
    address: Optional[str] = None
    is_active: bool = True
    created_at: Optional[str] = None


class VendorCreateRequest(BaseModel):
    """Payload to create a vendor."""

    name: str = Field(..., min_length=1)
    address: str = ""


class VendorUpdateRequest(BaseModel):
    """Payload to update a vendor."""

    name: str = Field(..., min_length=1)
    address: str = ""
